### PR TITLE
Feat(medusa): apply transform query usage

### PR DESCRIPTION
**Pull request by adrien2p**
### what

Migrate customer-groups and collections end point to use the new pipe pattern

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/collections/add-products.ts [hunk 1]
```diff
@@ -28,18 +29,17 @@ import { validator } from "../../../../utils/validator"
  *  "200":
  *    description: OK
  */
-export default async (req, res) => {
+export default async (req: Request, res) => {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/collections/add-products.ts:32]: question: Any particular reason as to why we are only typing the request obj?

**Review by olivermrbl — COMMENTED** [review]
Super nice work! Since it's only a partial implementation, can I get you to add a PR description of what controllers have been refactored to use the new pattern?

File: packages/medusa/src/api/routes/admin/collections/add-products.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/api/routes/admin/collections/add-products.ts:32]: Yeah i will add the description i just didn’t get the time to do all and thing i will stop there for that one 👍 and for the typings there no particular reason i will update that

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**